### DISCUSSION TITLE
Persistent cloud history

### DISCUSSION THREAD

**Discussion by zachlloyd** [warpdotdev/Warp#636] [Product ideas]
One feature we are considering is building is a persistent, searchable cloud history.

The idea is that your history would follow you across sessions and computers and be accessible and searchable anywhere, not just by command, but also by other metadata like output, time, user, machine, and more.

Use cases are finding and rerunning old commands, finding old output, sharing them with your team, auditing sessions.

We'd need to be super careful on the privacy side here, and would make this feature opt-in and possibly self-hosted.

Curious though what the community's reaction is.  Please upvote / downvote.

**Comment by koalazub** [comment]
Have there been any discussions on how you could make it private by default? Companies like [Skiff](https://www.skiff.org/) have a means to ensure that data is secure and owned by the users. And if that's maybe too 'out there' there might be some form of obfuscation and/or protections that companies like [Tresorit](tresorit.com) lean towards. 

I definitely agree that this is a serious privacy concern as developers can handle ultra sensitive information on a daily basis. So exploring a means to allow for a cloud history that's as secure as possible leans on imperative.

**Comment by elviskahoro** [comment]
Thanks for sharing these! @koalazub
